chia/wallet/cat_wallet/cat_wallet.py

```diff
@@ -213,6 +213,22 @@ async def get_or_create_wallet_for_cat(
 
         self.lineage_store = await CATLineageStore.create(self.wallet_state_manager.db_wrapper, self.get_asset_id())
         await self.wallet_state_manager.add_new_wallet(self)
+
+        delete: bool = False
+        for state in await self.wallet_state_manager.interested_store.get_unacknowledged_states_for_asset_id(
+            limitations_program_hash
+        ):
+            new_peer = self.wallet_state_manager.wallet_node.get_full_node_peer()
+            if new_peer is not None:
+                delete = True
+                peer_id: bytes32 = new_peer.peer_node_id
+                await self.wallet_state_manager.retry_store.add_state(state[0], peer_id, state[1])
```

Comment:
1) You might create the wallet directly instead of specifically adding it as an unacknowledged asset.  In this case, we still want to pick up the CATs that we've already synced but that we are just not showing to the user.  The core problem here is that CATs are special in that we sync them fully, we just at the end decide whether or not we want to keep them.  Previously, we would just forget them entirely, and when the wallet was added, we would resubscribe to a bunch of stuff and sync them again.  Now, we need some way to go back in time and resync the states that we skipped the first time.

2) I'm wary of directly calling the `add_coin_states` method outside of the normal process that calls it.  The wallet node has all of this careful task batching/cancelling and transactionality and whatnot that I don't fully understand, but basically the retry store is the easiest way to insert the coins to be processed in the normal order of things.  If someone makes changes to how coin states are processed in the future, I want to make sure it applies to these too and not create yet another entry point for coin states.